lib/cache.js

```diff
@@ -131,6 +162,54 @@ with --force.`)
     this.npm.output(`Index entries: ${stats.totalEntries}`)
     this.npm.output(`Finished in ${stats.runTime.total / 1000}s`)
   }
+
+  async searchCacheForPackage (pkg, ver) {
+    const cachePath = path.join(this.npm.cache, '_cacache')
+    const cacheList = await cacache.ls(cachePath)
+    const keyList = Object.keys(cacheList)
+    const output = []
+    if (pkg === undefined)
+      pkg = '(.*)'
+    const search = new RegExp(`^make-fetch-happen:request-cache:(.*)/${pkg}/-/(.*)-(.*).tgz`)
```

Comment:
This filename is more fraught with danger than you may realize. Did you know that semver prereleases can contain hyphens? It's fun!